sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -210,6 +210,7 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
   override def shutdown(): Unit = {
     GpuSemaphore.shutdown()
     PythonWorkerSemaphore.shutdown()
+    GpuDeviceManager.shutdown()
```

Comment:
I wonder if we should make these classes (Auto)Closable? then we could use hadoop's IOUtils 
to `IOUtils.cleanup(null, GpuSemaphore,  PythonWorkerSemaphore, GpuDeviceManager)`

Reply:
So we actually have `safeClose` already. `safeClose` will call `.close()` on every member of the collection, but then will throw at the end if there was a problem closing any member. I can certainly do this as part of this PR, the classes we care about are exactly these three.

Reply:
Agree not necessary for this PR.

Reply:
Ok lets skip it for now, we can follow up with a cleanup PR.